### PR TITLE
COMMERCE-10850 As a Developer I want to implement RelatedInfoItemCollectionProvider for Related Diagrams

### PR DESCRIPTION
**Pull request by alessiorendina**
Resent from #3543 

@MahmoudAzzam please check my changes, thanks! Diagrams should be visible only to DXP users.
@baharturk please re-test it manually and check if you need to change the functional test, thank you!

### CODE REVIEW

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by baharturk** [comment]
ci:test:commerce-functional

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce-functional

**Review by andrea-ale-sbarra — APPROVED** [review]

**Comment by alessiorendina** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceDisplayPageTemplates.testcase [hunk 1]
```diff
@@ -886,6 +886,125 @@ definition {
 		}
 	}
 
+	@description = "COMMERCE-9392. Verify that Related Diagrams are available for use with Collection Display Fragment."
+	@priority = 5
+	test CanMapRelatedDiagramsToCollectionDisplayFragment {
+		property portal.acceptance = "true";
+
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceDisplayPageTemplates.testcase:893]: @alessiorendina can you please add this here: 
`property test.run.environment = "EE";`
so we can prevent this test from running on CE environments. Thanks!

Comment (alessiorendina) [reply modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceDisplayPageTemplates.testcase:893]: Sure

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce-functional

**Comment by lorenzolapescara** [comment]
ci:test:relevant

**Comment by lorenzolapescara** [comment]
ci:stop:commerce-functional

**Pull request closed without merging**